support-frontend/assets/pages/[countryGroupId]/components/checkoutComponent.tsx

```diff
@@ -890,10 +926,39 @@ export function CheckoutComponent({
 								<CheckoutDivider spacing="loose" />
 							</>
 						)}
+						{deliveryPostcodeIsOutsideM25 && (
+							<FormSection>
+								<Legend>3. Delivery Agent</Legend>
+								<DeliveryAgentsSelect
+									chosenDeliveryAgent={chosenDeliveryAgent}
+									deliveryAgentsResponse={deliveryAgents}
+									setDeliveryAgent={(agent: number | undefined) => {
+										setChosenDeliveryAgent(agent);
+										setDeliveryAgentError(undefined);
+									}}
+									formErrors={
+										deliveryAgentError !== undefined
+											? [
+													{
+														field: 'deliveryProvider',
+														message: deliveryAgentError,
+													},
+											  ]
+											: []
+									}
+									deliveryAddressErrors={[]}
+								/>
+							</FormSection>
+						)}
 
 						<FormSection>
 							<Legend>
-								{productDescription.deliverableTo ? '3' : '2'}. Payment method
+								{productDescription.deliverableTo
+									? deliveryPostcodeIsOutsideM25
+										? '4'
+										: '3'
+									: '2'}
+								. Payment method
```

Comment:
The numbering logic gets a bit messy with two optional sections, I think it's readable enough though?